### DISCUSSION TITLE
TPS2113 outputs only 1V

### DISCUSSION THREAD

**Discussion by Krystman** [sanni/cartreader#898] [Need help]
I'm currently building my 3rd REV5 reader. I ran into a weird issue with this 3rd build. I'm using the Automatic Voltage Selection. But the Arduino won't power up. I poked around with the multimeter and noticed that the TPS2113 IC is outputting only around 1V on pin 7. It's getting 5V on pin 6 and 3V on pin 8 alright. I also checked continuity on all of the other pins multiple times. It is hooked up correctly. And fair enough, if I bypass the Automatic Voltage Selection and supply 3V on VCC the Arduino boots up correctly. Could this be an issue with the 0402 components or is this a faulty TPS2113? Or maybe something else?

**Comment by Ancyker** [comment]
Check that the chip has a proper ground. Also, check for any bridged pins. Make sure the Arduino/Firmware is configured for VSELECT (VSELECT uses the same pin as the LED, so failing to enable it causes the Arduino to try to PWM the TPS2113).

**Reply by Krystman** [reply]
I did check continuity to ground on both ground pins multiple times. I'll do another pass.
Configuration shouldn't be the issue. I did try to swap in an Arduino from a different, working build. Same result.

Which pin is the voltage IC wired to? I did use a stronger resistor on the LED because I didn't like how bright it was. Could this be the problem?

**Reply by Ancyker** [reply]
> Which pin is the voltage IC wired to? I did use a stronger resistor on the LED because I didn't like how bright it was.

https://github.com/sanni/cartreader/wiki/Automatic-Voltage-Selection#installation

> When using Vselect, you must not install the (...) status/power LED. (...) This is because when Vselect is used, the pin normally used for the status LED is instead used to toggle the voltage.

**Reply by Krystman** [reply]
Oh! Wow, yeah I completely missed that. Weirdly enough I have LEDs on my other two builds and they work fine. I now removed the LED and the resistor on the problematic build but it is still not working.

I'll try re-flowing the IC. If that doesn't work, I will replace the IC with a different one just to check if that was the issue.

**Reply by Ancyker** [reply]
> Weirdly enough I have LEDs on my other two builds and they work fine.

Are you sure? Voltage shouldn't switch when the LED is installed because it's normally configured as an open drain (the Arduino toggles it by making the pin GND). So, when you connect the LED you are connecting the toggle pin to the VCC rail, keeping it pulled up and always on.

**Reply by Ancyker** [reply]
Ignore that last post (that I deleted). I forgot it's toggling via VSNS. If you supply VSNS with VCC, which is what you do when you install the LED, then VSELECT should always output 3.3V. It should never toggle to 5V.

![image](https://github.com/sanni/cartreader/assets/7105333/6dde0792-a8e0-4b54-b31d-60ead417be4c)

_Edit_
Note: The datasheet is wrong. It says EN here when it means VSNS. This is why I was wrong in my other reply, I forgot the datasheet was incorrect. (I had originally discovered this out when I was prototyping VSELECT, but forgot in the months since.)

**Reply by Krystman** [reply]
> when you install the LED, then VSELECT should always output 3.3V. It should never toggle to 5V.

Well, here is what is happening in my two (now 3) working builds

The multimeter measures VCC. Here is what the OSCR boots into. 

![oscr_3v](https://github.com/sanni/cartreader/assets/10092987/6837edbe-219c-4cf1-abbc-1753d05ffee3)

And here is what happens when I chose to dump an original Game Boy cart

![oscr_5v](https://github.com/sanni/cartreader/assets/10092987/5ab396f8-ea5b-4e54-90eb-b9d73037d2f9)

Looks like it's working normally after all. The LED basically turns into a 5v indicator.

In the meantime, I've built 2 more boards. One works. One exhibits the issue described in the initial post. 

Also, on one of the two malfunctioning ones, I re-flowed the TPS2113. Took it off completely. Put it back in. No change. Will attempt an IC swap with a working model next.

**Reply by Ancyker** [reply]
When I tested it, I tested with a lower resistance value. I know you said you used a high-value resistor, so that might be why it works. What's the voltage to VSNS?

**Reply by Krystman** [reply]
Ok mystery solved. Swapped the TPS2113 from a working build to a malfunctioning one. That fixed the problem. I took the IC from the malfunctioning build and installed in on the board that previously worked. That board stopped working. This suggests to me that I got unlucky and got 2 bad ICs. Or maybe I damaged them somehow. Ordered more ICs.

>  I know you said you used a high-value resistor, so that might be why it works. What's the voltage to VSNS?

I used a 1K resistor. The voltage to VSNS is 3.3V when in 3.3V mode and the LED is off. It is close to 0V (0.03V) when in 5V mode and the LED is on. This is pretty much the same as without the LED.
